services/src/main/java/org/keycloak/protocol/oid4vc/issuance/OID4VCIssuerEndpoint.java

```diff
@@ -737,7 +755,24 @@ private CredentialRequest decryptCredentialRequest(String jweString, CredentialR
 
         // Parse decrypted content to CredentialRequest
         try {
-            return JsonSerialization.mapper.readValue(content, CredentialRequest.class);
+            CredentialRequest credentialRequest = JsonSerialization.mapper.readValue(content, CredentialRequest.class);
+
+            // Convert proof to proofs if proof is provided but proofs is not
+            if (credentialRequest.getProof() != null && credentialRequest.getProofs() == null) {
+                LOGGER.debugf("Converting single 'proof' field to 'proofs' array for backward compatibility");
+                Proof singleProof = credentialRequest.getProof();
+                Proofs proofsArray = new Proofs();
+                
+                // Convert single proof to proofs array format
+                if (singleProof.getJwt() != null) {
+                    proofsArray.setJwt(List.of(singleProof.getJwt()));
+                }
+                
+                credentialRequest.setProofs(proofsArray);
+                credentialRequest.setProof(null); // Clear the proof field to avoid confusion
+            }
```

Comment:
I would suggest to remove this duplicate code here. And do it just once in the validateRequestEncryptionMethod